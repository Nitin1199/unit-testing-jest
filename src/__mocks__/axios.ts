// __mocks__/axios.js
const axios = {
  get: jest.fn((url) => {
    if (url === 'https://jsonplaceholder.typicode.com/todos/1') {
      return Promise.resolve({id: 1, title: 'Test Todo', completed: false });
    } else if (url === 'https://jsonplaceholder.typicode.com/todos') {
      return Promise.resolve([
          { id: 1, title: 'Test Todo 1', completed: false },
          { id: 2, title: 'Test Todo 2', completed: true }
        ]);
    } else {
      return Promise.reject(new Error('Not Found'));
    }
  })
};

module.exports = axios;
