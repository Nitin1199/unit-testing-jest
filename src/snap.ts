import axios from 'axios';

export const todosUrl = 'https://jsonplaceholder.typicode.com/todos';
export const getTodoById = async (id: number) => {
  const result = await axios.get(`${todosUrl}/${id}`);
  return result;
};