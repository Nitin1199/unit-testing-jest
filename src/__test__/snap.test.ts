import axios from 'axios';
import { getTodoById } from '../snap';

describe('Snap Test Suite', () => {
  it('should create and test a snapshot', async () => {
    jest.spyOn(axios, 'get').mockResolvedValue({
      'userId': 2,
      'id': 1,
      'title': 'delectus aut autem',
      'completed': false
    });
    const todoId = 1;
    const result = await getTodoById(todoId);
    expect(result).toMatchSnapshot();
  });
});