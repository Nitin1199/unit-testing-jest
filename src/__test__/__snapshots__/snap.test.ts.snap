// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snap Test Suite should create and test a snapshot 1`] = `
{
  "completed": false,
  "id": 1,
  "title": "delectus aut autem",
  "userId": 2,
}
`;
